File: scripts/fixtures/pytest-sample/textkit/slug.py
"""Small text helpers — fixture for the testgen criterion run."""


def slugify(text: str) -> str:
    """Lowercase, trim, and join words with hyphens."""
    words = text.strip().lower().split()
    return "-".join(words)


def truncate(text: str, limit: int) -> str:
    """Cut text to limit characters, appending an ellipsis when cut."""
    if limit <= 0:
        return ""
    if len(text) <= limit:
        return text
    return text[: limit - 1] + "…"


def word_count(text: str) -> int:
    """Number of whitespace-separated words."""
    return len(text.split())

File: scripts/fixtures/vitest-sample/src/slug.ts
/** Small text helpers — fixture for the testgen criterion run. */

export function slugify(text: string): string {
  const words = text.trim().toLowerCase().split(/\s+/).filter((w) => w.length > 0);
  return words.join("-");
}

export function truncate(text: string, limit: number): string {
  if (limit <= 0) return "";
  if (text.length <= limit) return text;
  return text.slice(0, limit - 1) + "…";
}

export function wordCount(text: string): number {
  return text.split(/\s+/).filter((w) => w.length > 0).length;
}
